packages/preset-uno/src/rules/animation.ts

```diff
@@ -104,11 +104,69 @@ const keyframes: Record<string, string> = {
   'back-out-left': '{0%{opacity:1;transform:scale(1)}80%{opacity:0.7;transform:translateX(-2000px) scale(0.7)}100%{opacity:0.7;transform:translateY(-700px) scale(0.7)}}',
 }
 
+const durations: Record<string, string> = {
+  ...getDefaultValue(keyframes, '1s'),
+  'heart-beat': '1.3s',
+  'bounce-in': '0.75s',
+  'bounce-out': '0.75s',
+  'flip-out-x': '0.75s',
+  'flip-out-y': '0.75s',
+  'hinge': '2s',
+}
+
+const timingFns: Record<string, string> = {
+  ...getDefaultValue(keyframes, 'linear'),
+  'head-shake': 'ease-in-out',
+  'heart-beat': 'ease-in-out',
+  'pulse': 'ease-in-out',
+  'light-speed-in-left': 'ease-out',
+  'light-speed-in-right': 'ease-out',
+  'light-speed-out-left': 'ease-in',
+  'light-speed-out-right': 'ease-in',
+}
+
+const iterations: Record<string, string> = {
+  ...getDefaultValue(keyframes, 'infinite'),
+}
+
+const properties: Record<string, object | null> = {
+  ...getDefaultValue(keyframes, null),
+  'bounce': { 'transform-origin': 'center bottom' },
+  'jello': { 'transform-origin': 'center' },
+  'swing': { 'transform-origin': 'top center' },
+  'flip': { 'backface-visibility': 'visible' },
+  'flip-in-x': { 'backface-visibility': 'visible !important' },
+  'flip-in-y': { 'backface-visibility': 'visible !important' },
+  'flip-out-x': { 'backface-visibility': 'visible !important' },
+  'flip-out-y': { 'backface-visibility': 'visible !important' },
+  'rotate-in': { 'transform-origin': 'center' },
+  'rotate-in-down-left': { 'transform-origin': 'left bottom' },
+  'rotate-in-down-right': { 'transform-origin': 'right bottom' },
+  'rotate-in-up-left': { 'transform-origin': 'left bottom' },
+  'rotate-in-up-right': { 'transform-origin': 'right bottom' },
+  'rotate-out': { 'transform-origin': 'center' },
+  'rotate-out-down-left': { 'transform-origin': 'left bottom' },
+  'rotate-out-down-right': { 'transform-origin': 'right bottom' },
+  'rotate-out-up-left': { 'transform-origin': 'left bottom' },
+  'rotate-out-up-right': { 'transform-origin': 'right bottom' },
+  'hinge': { 'transform-origin': 'top left' },
+  'zoom-out-down': { 'transform-origin': 'center bottom' },
+  'zoom-out-left': { 'transform-origin': 'left center' },
+  'zoom-out-right': { 'transform-origin': 'right center' },
+  'zoom-out-up': { 'transform-origin': 'center bottom' },
+}
+
+function getDefaultValue<T>(map: Record<string, string>, defaultValue: T): Record<string, T> {
+  return Object.fromEntries(Object.entries(map).map(([key]) => ([key, defaultValue])))
+}
+
 export const animations: Rule[] = [
   [/^animate-(.*)$/, ([, name], { constructCSS }) => {
     const kf = keyframes[name]
-    if (kf)
-      return `@keyframes ${name}${kf}\n${constructCSS({ animation: `${name} 1s linear infinite` })}`
+    if (kf) {
+      return `@keyframes ${name}${kf}\n${constructCSS(
+        Object.assign({ animation: `${name} ${durations[name]} ${timingFns[name]} ${iterations[name]}` }, properties[name]))}`
```

Comment:
Could we assign the default value here instead of in the map to save some memory?

Meaning:
`${durations[name] || '1s'}`

Reply:
Of course